### PR TITLE
show piece captured in notation

### PR DESCRIPTION
**Pull request by Dillon-Roller**

### CODE REVIEW

File: chess/src/turn.rs [hunk 1]
```diff
@@ -4,13 +4,18 @@ use std::fmt;
 
 #[derive(Clone, Default)]
 pub struct Turn {
-    pub board_state: BoardState,
-    pub mv: Move,
+    pub(super) board_state: BoardState,
+    pub(super) mv: Move,
+    pub(super) piece_captured: bool,
```
Comment (DanielAlbl) [inline chess/src/turn.rs:9]: Nice, I like `super`

File: chess/src/history.rs [hunk 2]
```diff
@@ -59,7 +60,7 @@ impl History {
     }
 
     pub fn clone_current_state(&self) -> BoardState {
```
Comment (Dillon-Roller) [inline chess/src/history.rs:62]: i might just take this function out since it's `Copy` now

File: chess/src/history.rs [hunk 3]
```diff
@@ -37,17 +37,18 @@ impl History {
...
     pub fn add_info(&mut self, next_state: BoardState, mv: Move) {
-        *self.repetition_counter.entry(next_state).or_insert(0) += 1;
-        self.turns.push(Turn::new(next_state, mv));
-        self.current_turn += 1;
-        let previous_board_state = self.get_board_state(self.turns.len() - 1);
-        let is_pawn = previous_board_state.get_piece(&mv.from).unwrap().is_pawn();
-        let is_capture_move = previous_board_state.get_piece(&mv.to).is_some();
+        let current_state = self.get_current_state();
+        let is_pawn = current_state.get_piece(&mv.from).unwrap().is_pawn();
+        let is_capture_move = current_state.get_piece(&mv.to).is_some();
```
Comment (DanielAlbl) [inline chess/src/history.rs:42]: I think en passant is an exception here.

File: chess/src/game.rs [hunk 4]
```diff
@@ -336,14 +336,25 @@ impl Game {
...
-                        .get(1)
-                        .map_or(String::from("..."), |black| format!("{}", black)),
-                    i == self.get_current_round(),
-                )
+            .map(|(i, turns)| {
+                let mut white_string = String::default();
+                let mut black_string = String::default();
+
+                if turns[0].piece_captured {
+                    white_string.push('x')
```
Comment (DanielAlbl) [inline chess/src/game.rs:344]: Should this be in the `fmt` for `turn`?

File: chess/src/history.rs (hunk 2 above)
Comment (DanielAlbl) [reply chess/src/history.rs]: Oh yeah I forgot about that

File: chess/src/history.rs (hunk 3 above)
Comment (Dillon-Roller) [reply chess/src/history.rs:42]: En passant is a capture

Comment (DanielAlbl) [reply chess/src/history.rs:42]: I know and `get_piece(&mv.to)` would be `None` in that case

File: chess/src/game.rs [hunk 5]
```diff
@@ -92,7 +88,7 @@ impl Game {
         if let Some(piece) = self.get_piece(&from) {
             let mv = Move::new(from, to);
             self.is_move_valid(&mv)?;
-            let mut next_state = self.clone_current_state();
+            let mut next_state = *self.get_current_state();
```
Comment (DanielAlbl) [inline chess/src/game.rs:91]: Nice

**Review by DanielAlbl — DISMISSED** [review]

**Review by DanielAlbl — DISMISSED** [review]

**Review by DanielAlbl — APPROVED** [review]

**Pull request merged**